Modules/grpmodule.c

```diff
@@ -119,16 +120,56 @@ grp_getgrgid_impl(PyObject *module, PyObject *id)
         }
         Py_DECREF(py_int_id);
     }
+#ifdef HAVE_GETGRGID_R
+    Py_BEGIN_ALLOW_THREADS
+    int status;
+    long bufsize = sysconf(_SC_GETGR_R_SIZE_MAX);
```

Comment:
`sysconf(_SC_GETGR_R_SIZE_MAX)` can return -1. Check this and use some reasonable default in this case.

The type of bufsize should be Py_ssize_t, or the test for integer overflow below should be more complex.